app/javascript/components/cloud-volume-form/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './cloud-volume-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const CloudVolumeForm = ({ recordId }) => {
+  const [{ fields, initialValues, isLoading }, setState] = useState({ fields: [], isLoading: !!recordId });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+  const promise = useMemo(() =>
+    API.get('/api/providers?expand=resources&attributes=id,name,supports_block_storage&filter[]=supports_block_storage=true'), []);
+
+  const loadSchema = (appendState = {}) => ({ data: { form_schema: { fields } } }) => {
+    setState(state => ({
+      ...state,
+      ...appendState,
+      fields,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/providers/${recordId}`).then((initialValues) => {
+        promise.then(loadSchema({ initialValues, isLoading: false }));
+      });
+    }
+  }, []);
+
+  const onSubmit = (values) => {
```

Comment:
If we're going to use the `edit` hidden field mentioned above, you have to destructure it here and omit its value:
```js
const onSubmit = ({ edit: _edit, ...values }) => {
```